### PR TITLE
Fix Bug 1129624 - Refactor the function to get download links

### PR DESCRIPTION
**Pull request by jgmize**
I took @kyoshino's #2815 and rebased it against the current master, pushed it to this branch and then to demo5, then I took https://github.com/mozilla/mcom-tests/pull/394 and rebased it against the current mcom-tests master and pushed it to https://github.com/mozilla/mcom-tests/tree/bug-1129624-pr-394 and ran that branch against demo5 in a new job on the Jenkins server we originally set up for mffos, and everything passed: https://ci.masterfirefoxos.com/view/www.mozilla.org/job/www-demo5.allizom.saucelabs-bug-1129624/14/console

### CODE REVIEW

**Comment by jgmize** [comment]
@pmclanahan I know you already gave the r+ on the code on #2815, and based on the testing I described earlier I'm pretty confident that I handled the merge conflicts correctly, but I'd appreciate one final r? on this and the associated https://github.com/mozilla/mcom-tests/pull/394 which we'll need to merge once this lands.

**Comment by pmclanahan** [comment]
@jgmize I did. Did you take a look as well. It's a big change, but I feel a good one. With the tests passing I'm feeling good about it not introducing regressions. So if you're pleased, I'll take another look now and hopefully merge today.

File: bedrock/firefox/helpers.py [hunk 1]
```diff
@@ -1,9 +1,194 @@
...
+        for type, arch_pretty in variations.iteritems():
+            link = firefox_android.get_download_url('alpha', type)
+            builds.append({'os': 'android',
+                           'os_pretty': 'Android',
+                           'os_arch_pretty': 'Android %s' % arch_pretty,
+                           'arch': 'x86' if type == 'x86' else 'armv7 %s' % type,
+                           'arch_pretty': arch_pretty,
+                           'download_link': link})
+
+    if channel != 'alpha':
```
Comment (jgmize) [inline bedrock/firefox/helpers.py]: NIT: I think `else` would be more readable

**Comment by jgmize** [comment]
@pmclanahan I just went over it again, found a tiny [readability nit](https://github.com/mozilla/bedrock/pull/2871#discussion_r27406709) and went ahead and fixed it.

**Comment by jgmize** [comment]
Yes, I think we're good to go here.

**Comment by pmclanahan** [comment]
Cool. Agreed. :shipit:

**Comment by kyoshino** [comment]
:clap:

**Comment by pmclanahan** [comment]
Thanks @kyoshino ! :+1:

**Comment by kyoshino** [comment]
Yay!

**Pull request merged**